src/bokeh/models/textures.py

```diff
@@ -80,9 +84,14 @@ class ImageURLTexture(Texture):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
 
-    url = Required(String, help="""
+    url = Required(Image, help="""
     A URL to a drawable resource like image, video, etc.
 
+    If provided with a file path, the file will be encoded using ``data:``
+    protocol (utf-8 encoding for ``*.svg`` and base64 for ``*.png`` and
+    other binary formats).
+
+    NumPy 2D arrays are also supported and use ``data:`` encoding as well.
```

Comment:
FWIW suggest we deprecate `url` soon and eventually replace with a better name that actually encompasses that images are also valid values